lib/sles4sap_publiccloud_basetest.pm

```diff
@@ -28,6 +29,20 @@ sub cleanup {
         # Skip cleanup if ansible inventory is not present (deployment could not have been done without it)
         next if (script_run 'test -f ' . qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER')));
 
+
+        if (is_azure && check_var('IS_MAINTENANCE', 1)) {
```

Comment:
It is not ok to have this within the `for my $command ('ansible', 'terraform') {` loop. It can be done before the loop.